codec-http/src/main/java/io/netty/handler/codec/http/HttpContentEncoder.java

```diff
@@ -319,4 +327,22 @@ public EmbeddedChannel contentEncoder() {
             return contentEncoder;
         }
     }
+
+    private static final class RequestMetaData {
```

Comment:
@trustin It does complicate the logic as one is loosing the actual http method information, e.g.
httpMethod == HttpMethod.HEAD || (httpMethod == HttpMethod.CONNECT && code == 200);

one solution would be to extend the singleton CharSequence idea to indicate the actual method itself, HEAD or CONNECT? that implies text comparisons which would be more expensive than a reference check.